src/pmhn/_trees/_backend_geno.py

```diff
@@ -0,0 +1,236 @@
+from typing import Protocol, Optional
+
+import numpy as np
+from pmhn._trees._interfaces import Tree
+from pmhn._trees._tree_utils_geno import create_mappings
+from anytree import Node
+
+
+class LoglikelihoodSingleTree:
+    def __init__(self, tree: Node):
+        (
+            self._genotype_subtree_node_map,
+            self._index_subclone_map,
+        ) = create_mappings(tree)
+
+    _genotype_subtree_node_map: dict[tuple[tuple[Node, int]], tuple[int, int]]
```

Comment:
Id' move these to the `__init__` function (see the comment on class-level versus instance-level attributes.)